### PR TITLE
[Distributed] better diagnostics for missing actor system typealias

### PR DESCRIPTION
**Pull request by ktoso**
Thank you @kavon for reminding me about the fact that the failure is pretty ugly.

Resolves rdar://89585848 Tailored diagnostic for DA conformance fail

### CODE REVIEW

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -5391,6 +5391,32 @@ void swift::diagnoseConformanceFailure(Type T,
...
+    if (!nominal)
+      return;
+
+    // If it is missing the ActorSystem type, suggest adding it:
+    // FIXME(distributed): use getDistributedActorSystemType(nominal); once merged
+    auto &C = nominal->getASTContext();
+    auto DA = C.getDistributedActorDecl();
+    Type selfType = nominal->getSelfInterfaceType();
+    auto conformance = nominal->getParentModule()->lookupConformance(selfType, DA);
+    auto systemTy = conformance.getTypeWitnessByName(selfType, C.Id_ActorSystem);
```
Comment (ktoso) [inline lib/Sema/TypeCheckProtocol.cpp:5410]: I have this logic in the new DistributedDecl.cpp but ripping it out is a large change coming in a different PR...

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: include/swift/AST/DiagnosticsSema.def [hunk 2]
```diff
@@ -4485,6 +4485,12 @@ ERROR(actor_inheritance,none,
 ERROR(actor_protocol_illegal_inheritance,none,
       "non-actor type %0 cannot conform to the 'Actor' protocol",
       (DeclName))
+ERROR(distributed_actor_conformance_missing_system_type,none,
+      "distributed actor %0 does not declare ActorSystem it can be used with. "
+      "Declare the associated actor system with a 'typealias ActorSystem = MyActorSystem' "
```
Comment (xedin) [inline include/swift/AST/DiagnosticsSema.def:4490]: I'd suggest you split this into a note + fix-it to insert `typealias ActorSystem = <#Actor System#>` instead another node with talks about the default typealias and puts it right before the declaration.

Comment (ktoso) [reply include/swift/AST/DiagnosticsSema.def:4490]: Sounds good, will do

**Comment by ktoso** [comment]
Will rebase this after https://github.com/apple/swift/pull/41616 lands

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

**Pull request merged**